tests/test_assets.py

```diff
@@ -966,6 +970,198 @@ def test_sids(self):
         ))
         self.assertEqual({0, 1, 2}, set(self.asset_finder.sids))
 
+    def test_lookup_by_supplementary_mapping(self):
+        equities = pd.DataFrame.from_records(
+            [
+                {
+                    'sid': 0,
+                    'symbol': 'A',
+                    'start_date': pd.Timestamp('2013-1-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                    'exchange': 'TEST',
+                },
+                {
+                    'sid': 1,
+                    'symbol': 'B',
+                    'start_date': pd.Timestamp('2013-1-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                    'exchange': 'TEST',
+                },
+                {
+                    'sid': 2,
+                    'symbol': 'C',
+                    'start_date': pd.Timestamp('2013-7-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                    'exchange': 'TEST',
+                },
+            ]
+        )
+
+        supplementary_mappings = pd.DataFrame.from_records(
+            [
+                {
+                    'sid': 0,
+                    'mapping_type': 'ALT_ID',
+                    'value': '100000000',
+                    'start_date': pd.Timestamp('2013-1-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2013-6-28', tz='UTC'),
+                },
+                {
+                    'sid': 1,
+                    'mapping_type': 'ALT_ID',
+                    'value': '100000001',
+                    'start_date': pd.Timestamp('2013-1-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                },
+                {
+                    'sid': 0,
+                    'mapping_type': 'ALT_ID',
+                    'value': '100000002',
+                    'start_date': pd.Timestamp('2013-7-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                },
+                {
+                    'sid': 2,
+                    'mapping_type': 'ALT_ID',
+                    'value': '100000000',
+                    'start_date': pd.Timestamp('2013-7-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                },
+            ]
+        )
+
+        self.write_assets(
+            equities=equities,
+            supplementary_mappings=supplementary_mappings,
+        )
+
+        af = self.asset_finder
+
+        # Before sid 0 has changed ALT_ID.
+        dt = pd.Timestamp('2013-6-28', tz='UTC')
+
+        asset_0 = af.lookup_by_supplementary_mapping('ALT_ID', '100000000', dt)
```

Comment:
Yeah definitely agree, what do you think of `lookup_by_supplementary_field`, to go along with `get_supplementary_field`?